controllers/object_controls.go

```diff
@@ -712,6 +718,10 @@ func preProcessDaemonSet(obj *appsv1.DaemonSet, n ClusterPolicyController) error
 		return err
 	}
 
+	// transform the host-root and host-dev-char volumes if a custom host root is configured with the operator
+	transformHostRootVolume(obj, n.singleton.Spec.HostPaths.Root)
+	transformHostDevCharVolume(obj, n.singleton.Spec.HostPaths.Root)
```

Comment:
Does it make sense to have a single function `transformForHostRoot` here and then call these two in that function?

Reply:
Makes sense. I have added a `transformForHostRoot` function.